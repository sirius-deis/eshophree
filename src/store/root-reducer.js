import { combineReducers } from "redux";

import categoryReducer from "./category/category.reducer";
import userReducer from "./user/user.reducer";
import cartReducer from "./cart/cart.reducer";

const rootReducer = combineReducers({
    category: categoryReducer,
    user: userReducer,
    cart: cartReducer,
});

export default rootReducer;
