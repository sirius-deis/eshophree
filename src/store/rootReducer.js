import { combineReducers } from 'redux';

import userReducer from './user/user.reducer';
import categoryReducer from './category/category.reducer';

const rootReducer = combineReducers({
  user: userReducer,
  category: categoryReducer,
});

export default rootReducer;
