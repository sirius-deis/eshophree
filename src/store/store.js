import { createStore, applyMiddleware } from 'redux';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

import rootReducer from './rootReducer';

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const middlewares = [thunk];

if (process.env.NODE_ENV === 'development') {
  middlewares.push(logger);
}

export const store = createStore(persistedReducer, applyMiddleware(...middlewares));

export const persistor = persistStore(store);
