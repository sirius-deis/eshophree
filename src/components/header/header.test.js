import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store/store';
import { MemoryRouter } from 'react-router-dom';

import Header from './header';

describe('Header component', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
        ,
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
