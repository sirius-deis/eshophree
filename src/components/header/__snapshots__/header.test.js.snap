// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<div>
  <header
    class="sc-bgqQcB hFxHjZ"
  >
    <div
      class="container"
    >
      <div
        class="sc-gTRrQi WDvOc"
      >
        <a
          href="/"
        >
          <img
            alt="Logo"
            height="auto"
            src="logo.png"
            width="120"
          />
        </a>
        <form
          class="sc-ewnqHT bnBpQm"
        >
          <div
            class="sc-kcuKUB hZPXkU"
          >
            <button
              class="sc-iVCKna sc-kPTPQs ilBmZw jJkXSR"
              style="background-color: transparent;"
              type="button"
            >
              <div
                class="sc-dicizt cdbMzz"
              >
                All categories
              </div>
              <svg
                fill="currentColor"
                font-size="22"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </button>
          </div>
          <input
            class="sc-fFGjHI cfILGU"
            type="text"
          />
          <button
            class="sc-iVCKna sc-gxYJeL ilBmZw iRMDmu"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </button>
        </form>
        <div
          class="sc-duSInm joIcmp"
        >
          <div
            class="sc-kcuKUB hZPXkU"
          >
            <button
              class="sc-jrrXlR jShtjL"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                />
              </svg>
              <div
                class="sc-crozmw biEPil"
              >
                0
              </div>
            </button>
            <div
              class="sc-dcrmVg bABvsQ"
            />
          </div>
          <div
            class="sc-kcuKUB hZPXkU"
          >
            <button
              class="sc-jrrXlR jShtjL"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <div
                class="sc-crozmw biEPil"
              >
                0
              </div>
            </button>
            <div
              class="sc-dcrmVg bABvsQ"
            />
          </div>
          <div
            class="sc-kcuKUB hZPXkU"
          >
            <button
              class="sc-jrrXlR jShtjL"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M313.6 304c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 304 0 364.2 0 438.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-25.6c0-74.2-60.2-134.4-134.4-134.4zM400 464H48v-25.6c0-47.6 38.8-86.4 86.4-86.4 14.6 0 38.3 16 89.6 16 51.7 0 74.9-16 89.6-16 47.6 0 86.4 38.8 86.4 86.4V464zM224 288c79.5 0 144-64.5 144-144S303.5 0 224 0 80 64.5 80 144s64.5 144 144 144zm0-240c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
                />
              </svg>
            </button>
            <div
              class="sc-dcrmVg bABvsQ"
            />
          </div>
        </div>
      </div>
    </div>
    <nav
      class="sc-jJTsDX hGRRUE"
    >
      <div
        class="container"
      >
        <ul
          class="sc-cJPUEC cNMxCE"
        >
          <li
            class="sc-jeCNp jqhsJP"
          >
            <a
              href="/home"
            >
              home
            </a>
          </li>
          <li
            class="sc-jeCNp jqhsJP"
          >
            <a
              href="/categories"
            >
              categories
            </a>
          </li>
          <li
            class="sc-jeCNp jqhsJP"
          >
            <a
              href="/shop"
            >
              shop
            </a>
          </li>
          <li
            class="sc-jeCNp jqhsJP"
          >
            <a
              href="/about"
            >
              about
            </a>
          </li>
          <li
            class="sc-jeCNp jqhsJP"
          >
            <a
              href="/blog"
            >
              blog
            </a>
          </li>
          <li
            class="sc-jeCNp jqhsJP"
          >
            <a
              href="/contact"
            >
              contact
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
  ,
</div>
`;
